src/explorer.api/Explorers/DiffixUtilities.cs

```diff
@@ -0,0 +1,42 @@
+namespace Explorer
+{
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Linq;
+
+    using Explorer.Queries;
+
+    internal static class DiffixUtilities
+    {
+        internal static List<decimal> EstimateBucketResolutions(
+            long numSamples,
+            dynamic minSample,
+            dynamic maxSample,
+            long valuesPerBucketTarget)
+        {
+            Debug.Assert(numSamples > 0, "Argument numSamples should always be greater than zero.");
+
+            double valueDensity;
+            try
+            {
+                valueDensity = (double)numSamples / (maxSample - minSample);
+            }
+            catch (System.DivideByZeroException)
```

Comment:
You removed the division check (and added an assert instead). Will this be OK for the release version? Is there any reason why the min and max samples will be always different.

Reply:
Good point... I think it might be possible to have min == max if all the column values are the same. In this case, it makes more sense to return a default bucket size of, say, 1, rather than throw an exception. 

Also for the `numSamples > 0` check, it might be better to throw an exception rather than use an assertion. 

I'll make these changes.

Reply:
... done: a473728807be5ef6ce334732cf97ebeb1d06f733